lib/core/client.js

```diff
@@ -909,7 +909,10 @@ function _resume (client, sync) {
       request.body = null
     }
 
-    if (client.running && util.isStream(request.body)) {
+    if (client.running &&
```

Comment:
@ronag how do I get the request into a "running" state while also still having the request.body defined?

Reply:
Create two request with a non empty stream body?

Reply:
haven't forgotten about this. I tried again but didn't make much progress. I'm going to try another way later by manually stepping through the execution and seeing what I will need to do to get the code down a certain path

Reply:
I think I gotta ask for help on this. Really struggling to write tests to get to some of the edge cases in the writeIterable method. @ronag would you like to pair on this? otherwise, if you want to author the necessary tests that is okay with me

Reply:
I'll get back to you.

Reply:
@Ethan-Arrowood We already have tests which hit this when using streams? Can't you just find the test and replace the stream part with an iterable?

Reply:
yeah thats what I thought i must've introduce some regressions cause the coverage reporter was saying some of these lines were not getting tested.

Reply:
This needs significantly more tests. Preferably we should just port most of the body as stream test into a body as iterable test.